app/src/main/java/de/westnordost/streetcomplete/quests/construction/MarkCompletedBuildingConstruction.java

```diff
@@ -0,0 +1,74 @@
+package de.westnordost.streetcomplete.quests.construction;
+
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.util.Log;
+
+import java.util.Map;
+
+import javax.inject.Inject;
+
+import de.westnordost.osmapi.map.data.BoundingBox;
+import de.westnordost.streetcomplete.R;
+import de.westnordost.streetcomplete.data.meta.OsmTaggings;
+import de.westnordost.streetcomplete.data.osm.changes.StringMapChangesBuilder;
+import de.westnordost.streetcomplete.data.osm.download.MapDataWithGeometryHandler;
+import de.westnordost.streetcomplete.data.osm.download.OverpassMapDataDao;
+import de.westnordost.streetcomplete.data.osm.tql.OverpassQLUtil;
+import de.westnordost.streetcomplete.quests.DateHandler;
+import de.westnordost.streetcomplete.quests.YesNoQuestAnswerFragment;
+
+public class MarkCompletedBuildingConstruction extends MarkCompletedConstruction
+{
+	@Inject
+	public MarkCompletedBuildingConstruction(OverpassMapDataDao overpassServer) {
+		super(overpassServer);
+	}
+
+	@Override public boolean download(BoundingBox bbox, MapDataWithGeometryHandler handler)
+	{
+		return overpassServer.getAndHandleQuota(getOverpassQuery(bbox), handler);
+	}
+
+	/** @return overpass query string to get streets marked as under construction but excluding ones
+	 * - with tagged opening date that is in future
+	 * - recently edited (includes adding/updating check_date tags)
+	 * . */
+	private static String getOverpassQuery(BoundingBox bbox)
+	{
+		String groupName = ".buildings_under_construction";
+		return OverpassQLUtil.getGlobalOverpassBBox(bbox) +
+			getQueryPart("building", groupName, 180) +
+			groupName + " out meta geom;";
+	}
+
+	public void applyAnswerTo(Bundle answer, StringMapChangesBuilder changes)
+	{
+		if(answer.getBoolean(YesNoQuestAnswerFragment.ANSWER)) {
+			String constructionValue = changes.getPreviousValue("construction");
+			if(constructionValue == null) {
+				constructionValue = "yes";
+			}
+			changes.modify("building", constructionValue);
+			changes.deleteIfPresent("construction");
+			changes.deleteIfPresent("opening_date");
+			changes.deleteIfPresent(OsmTaggings.SURVEY_MARK_KEY);
+		} else {
+			changes.addOrModify(OsmTaggings.SURVEY_MARK_KEY, DateHandler.getCurrentDateString());
+		}
```

Comment:
Do you think it would make sense if these lines (53-58) would move to the super class?

Reply:
Moved back to superclass.